geometric/run_json.py

```diff
@@ -36,13 +36,18 @@ def parse_input_json_dict(in_json_dict):
         "program": "rdkit"
     }
     """
-    input_opts = in_json_dict['keywords'].copy()
-    input_specification = in_json_dict['input_specification'].copy()
+
+    in_json_dict = copy.deepcopy(in_json_dict)
+    input_opts = in_json_dict['keywords']
+    input_specification = in_json_dict['input_specification']
+
     # insert `fix_orientation` and `fix_com`
+    input_specification['molecule'] = copy.deepcopy(in_json_dict['initial_molecule'])
```

Comment:
Yes, `molecule` is the state that we update throughout the process. We do not want `initial_molecule` to be updated.

Reply:
I understand it's convenient to have `deepcopy` in multiple places, but if I understand correctly, this one is redundant because after the first `deepcopy` in line 40, `in_json_dict` is separated from the input parameter, and completely lives in the scope of this `parse_input_json_dict()` function. Therefore a second `deepcopy` on the `in_json_dict['initial_molecule']` doesn't seem to have any effect. Is that right?

Reply:
The `input_specification` in its entirety is passed into the engines. Currently, `QCEngineAPI` correctly handles the copies; however, this is not always guaranteed. The copy is very cheap and will likely catch quite a few errors in the future rather than keeping a mutable link.

Reply:
Hypothetically if we remove the second `deepcopy` here, and `QCEngineAPI` changes `input_specification['molecule']`, then the `in_json_dict['initial_molecule']` will also be changed, is that what you mean? I think since `in_json_dict` is already separated from the input parameter, changing `input_specification['molecule']` will not affect anything, because the temporary copy of `in_json_dict` inside this function will be garbage collected right after this line.

Reply:
Ah, your right I forgot that I had switched from `.copy()` which *is* bad to `deepcopy` early on.